### PR TITLE
feat: add strict versioned API autocomplete MONGOSH-770

### PR DESCRIPTION
**Pull request by addaleax**

### CODE REVIEW

File: packages/autocomplete/index.ts [hunk 1]
```diff
@@ -217,14 +224,23 @@ function filterShellAPI(
     if (!c.startsWith(prefix)) return false;
     if (completions[c].deprecated) return false;
 
-    const serverVersion = params.connectionInfo()?.server_version;
-    if (!serverVersion) return true;
+    const apiVersionInfo = params.apiVersionInfo();
+    let isAcceptableVersion;
+    let acceptableApiVersions;
+    if (apiVersionInfo?.strict && (acceptableApiVersions = completions[c].apiVersions)) {
```
Comment (rose-m) [inline packages/autocomplete/index.ts:230]: I would strongly prefer not having the assignment as a side effect in there if you don't mind 😅 
```suggestion
    if (apiVersionInfo?.strict && completions[c].apiVersions) {
    	const acceptableApiVersions = completions[c].apiVersion;
```

**Review by rose-m — COMMENTED** [review]
So nice how the decorators make it easy to implement that 👍

File: packages/autocomplete/index.ts (hunk 1 above)
Comment (addaleax) [reply packages/autocomplete/index.ts:230]: Yeah, this is exactly what I had here before, character-by-character :) TypeScript doesn’t get that the first `completions[c].apiVersions` being truthy implies that `acceptableApiVersions` is also set, unfortunately…

Comment (rose-m) [reply packages/autocomplete/index.ts:230]: Ah crap... go ya... I'm fine with leaving it like that then 👍

**Review by rose-m — APPROVED** [review]

**Pull request merged**